pkg/cli/admin/release/extract.go

```diff
@@ -349,6 +349,28 @@ func (o *ExtractOptions) Run(ctx context.Context) error {
 		}
 	}
 
+	// versionInImageConfig stores the version from the label of the image
+	// At the moment it is used later only for the logging purpose and thus is not blocking if failures occur upon getting its value
+	var versionInImageConfig string
+	opts.ImageConfigCallback = func(imageConfig *dockerv1client.DockerImageConfig) {
+		if imageConfig == nil {
+			// This should never happen
+			klog.Error("Cannot retrieve the version because no image configuration is provided in the image to extract")
+			return
+		}
+		if imageConfig.Config == nil {
+			klog.Error("Cannot retrieve the version from image configuration in the image to extract because it has no configuration")
+			return
+		}
+		v, ok := imageConfig.Config.Labels["io.openshift.release"]
+		if !ok {
+			klog.Error("Cannot retrieve the version from image configuration in the image to extract because it does not have the required label 'io.openshift.release'")
+			return
+		}
+		klog.V(2).Infof("Retrieved the version from image configuration in the image to extract: %s", v)
+		versionInImageConfig = v
```

Comment:
We assume we never reach this place twice here (= we assume the callback is never called multiple time in a way that reaches this assignment) otherwise the old value is overwritten.

That assumption may be fine but I'd either add a comment here about it or logged if we reach here while `versionInImageConfig` is not empty.